index.ts

```diff
@@ -155,3 +159,48 @@ function getRepoCredentials(projectUrl: string) {
     const repoName = urlPath[2];
     return [ownerName, repoName];
 }
+
+/**
+ * Add a assignee on bounties that are currently assigned
+ * @param devpoolIssues list of all devpool issues
+ */
+
+function assignAssigneeToUnavailableIssues(devpoolIssues: Issue[]){
+  devpoolIssues.map(async (item) => {
+    //get owner name , reponame and issue name from a link of the issue
+    const [ownerName, repoName, issueNumber] = getIssueDetailsWithLink(
+      item?.body ?? ""
+    );
+    //get the issue details
+    const issueDetails = await octokit.rest.issues.get({
+      owner: ownerName,
+      repo: repoName,
+      issue_number: parseInt(issueNumber, 10),
+    });
+    //check if there is an assignie to the issue and update the issue
+    if (issueDetails.data?.assignee) {
+      await octokit.rest.issues.update({
+        owner: DEVPOOL_OWNER_NAME,
+        repo: DEVPOOL_REPO_NAME,
+        issue_number: item.number,
+        assignees:[issueDetails.data?.assignee.login]
+      });
+      console.log(`Updated: ${item.html_url}`);
+    }
+  });
+};
+
+/**
+ * Returns owner, repository names and issues number from a issue URL
+ * @param projectUrl issue URL
+ * @returns array of owner , repository names and issue number
+ */
+
+function getIssueDetailsWithLink(projectUrl: string) {
```

Comment:
This method https://github.com/ubiquity/devpool-directory/blob/9a87b40bff54c5dc81f0205faf79f18c0ba2fba9/index.ts#L155 does pretty much the same job, pls refactor `getRepoCredentials()` and remove `getIssueDetailsWithLink ()`

Reply:
done